lib/events/split_events_listener.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:splitio/events/split_method_call_handler.dart';
+import 'package:splitio/split_client.dart';
+
+abstract class SplitEventsListener {
+  Future<SplitClient> onReady();
```

Comment:
is `onXXX` the right naming convention for functions that returns a promise/future? or should it be `whenXXX`?